teps/0048-pipeline-as-code.md

```diff
@@ -0,0 +1,214 @@
+if it was succesfull or failed.
+
+With this initial proposal we have focused on a PR based workflow and not on the
+post-merge workflows. But there is nothing in this design that can't
+address post-merge workflows.
+
+We are not addressing here on how to check out private repositories and how to
+get token from the Web VCS provider (ie: via Github or Gitlab [personal
+token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token)
+or via an [Oauth
+application](https://docs.github.com/en/free-pro-team@latest/developers/apps))
+
+### Risks and Mitigations
+
+
+#### Logs retention
+
+There is currently no way with Tekton to have the logs in different locations
+which means we assume the logs will be kept as long as the `Pipelinerun` is
+still in the cluster.
+
+### User Experience
+
+* A user deploy with a single command the pipelines as code template and retrieve
+  the ingress url of the eventlistenner webhook url.
+* The user configure the webhook URL in GitHub or other web vcs provider.
+* A user add a tekton template in the `.tekton` directory of its repository.
+* User sends a PR.
+* Pipeline as code start applying the template located in the `.tekton`
+  directory
+* User is able to watch the pipeline execution status via `tkn` or `dashboard`.
+* Pipeline as code, succeed or fail accordingly.
+
+### Performance (optional)
+
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+## Design Details
+
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+Instead of having a Tekton as a code task with a binary doing the 'work' we may
+want to do this directly in triggers.
+
+This is something that is possible to do currently but from my experience working
+on POC and the future
```

Comment:
We will need more detailled alternative approach here at some point

Reply:
What kind of details would you like to see ? Alternative implementation or alternative design achieving tekton as code ?

The alternative design is I guess an ad-hoc approach, like the way we do right know which is not really tekton as code i guess.